In [1]:
import pandas as pd

This is the website for all CFPB complaints: https://www.consumerfinance.gov/data-research/consumer-complaints/search/?company=Social%20Finance%2C%20Inc.&date_received_max=2023-05-20&date_received_min=2011-12-01&page=1&searchField=all&searchText=social%20finance%2C%20inc.&size=25&sort=created_date_desc&tab=List

In [2]:
#Import csv file with all complaints on CFPB website
df = pd.read_csv("complaints-2023-05-20_15_56.csv")
df

Date received                                            Product  \
0        05/05/19                                       Student loan   
1        06/06/22                        Checking or savings account   
2        08/11/21                                           Mortgage   
3        10/21/21          Payday loan, title loan, or personal loan   
4        06/11/19                                    Debt collection   
..            ...                                                ...   
987      05/20/22                        Checking or savings account   
988      05/24/22  Credit reporting, credit repair services, or o...   
989      10/09/17                                       Student loan   
990      04/05/22  Credit reporting, credit repair services, or o...   
991      10/17/17  Credit reporting, credit repair services, or o...   

                        Sub-product  \
0              Private student loan   
1                  Checking account   
2        Conventional home mortgage   
3           Personal line of credit   
4         Private student loan debt   
..                              ...   
987                Checking account   
988                Credit reporting   
989  Federal student loan servicing   
990                Credit reporting   
991                Credit reporting   

                                                 Issue  \
0                                       Getting a loan   
1                                   Opening an account   
2                       Trouble during payment process   
3                             Getting a line of credit   
4                    Attempts to collect debt not owed   
..                                                 ...   
987                                Managing an account   
988  Problem with a credit reporting company's inve...   
989                      Struggling to repay your loan   
990               Incorrect information on your report   
991                        Improper use of your report   

                                             Sub-issue  \
0                                          Denied loan   
1            Didn't receive terms that were advertised   
2                                                 None   
3                                                 None   
4                                        Debt was paid   
..                                                 ...   
987                           Deposits and withdrawals   
988  Their investigation did not fix an error on yo...   
989             Problem lowering your monthly payments   
990                Information belongs to someone else   
991  Credit inquiries on your report that you don't...   

                          Consumer complaint narrative  \
0    Hello, I applied for a student loan refinance ...   
1                                                  NaN   
2    Mortgage was paid off on XX/XX/2021 which resu...   
3                                                  NaN   
4    I refinanced my loan with Sofi through XXXX XX...   
..                                                 ...   
987  Hi, I opened an account with SoFi recently, an...   
988                                                NaN   
989  The CFPB has taken action against numerous ban...   
990                                                NaN   
991                                                NaN   

    Company public response               Company State ZIP code  \
0                      None  Social Finance, Inc.    CT    06854   
1                      None  Social Finance, Inc.    CA    94110   
2                      None  Social Finance, Inc.    FL    32223   
3                      None  Social Finance, Inc.    MD    21237   
4                      None  Social Finance, Inc.    CA    91320   
..                      ...                   ...   ...      ...   
987                    None  Social Finance, Inc.    NJ    077XX   
988                    None  Social

In [4]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 992 entries, 0 to 991
Data columns (total 18 columns):
 #   Column                        Non-Null Count  Dtype 
---  ------                        --------------  ----- 
 0   Date received                 992 non-null    object
 1   Product                       992 non-null    object
 2   Sub-product                   992 non-null    object
 3   Issue                         992 non-null    object
 4   Sub-issue                     992 non-null    object
 5   Consumer complaint narrative  591 non-null    object
 6   Company public response       992 non-null    object
 7   Company                       992 non-null    object
 8   State                         992 non-null    object
 9   ZIP code                      992 non-null    object
 10  Tags                          992 non-null    object
 11  Consumer consent provided?    903 non-null    object
 12  Submitted via                 992 non-null    object
 13  Date sent to company

In [6]:
df["Consumer disputed?"].unique()

array([nan, 'No', 'Yes'], dtype=object)

In [9]:
dfY = df.query("`Consumer disputed?` == 'Yes'")
dfY

Date received                  Product               Sub-product  \
207      11/24/15            Consumer Loan          Installment loan   
410      11/09/16             Student loan  Non-federal student loan   
426      03/02/16             Student loan  Non-federal student loan   
436      12/07/16             Student loan  Non-federal student loan   
481      03/13/14             Student loan  Non-federal student loan   
564      06/03/16            Consumer Loan          Installment loan   
593      02/24/17            Consumer Loan          Installment loan   
633      02/04/16            Consumer Loan          Installment loan   
634      08/29/16  Other financial service           Debt settlement   
637      11/17/16             Student loan  Non-federal student loan   
643      10/02/15             Student loan  Non-federal student loan   
645      10/23/16            Consumer Loan          Installment loan   
657      08/27/15             Student loan  Non-federal student loan   
787      03/19/16            Consumer Loan          Installment loan   
811      02/02/16             Student loan  Non-federal student loan   
826      11/18/14             Student loan  Non-federal student loan   
834      10/01/16             Student loan  Non-federal student loan   
873      01/04/17            Consumer Loan          Installment loan   
917      01/04/16            Consumer Loan          Installment loan   
922      11/15/16            Consumer Loan          Installment loan   
965      05/14/15            Consumer Loan          Installment loan   

                                  Issue  \
207        Taking out the loan or lease   
410  Dealing with my lender or servicer   
426  Dealing with my lender or servicer   
436  Dealing with my lender or servicer   
481                      Getting a loan   
564        Taking out the loan or lease   
593        Taking out the loan or lease   
633        Taking out the loan or lease   
634           Advertising and marketing   
637                      Getting a loan   
643                      Getting a loan   
645        Taking out the loan or lease   
657                      Getting a loan   
787        Taking out the loan or lease   
811                      Getting a loan   
826                      Getting a loan   
834                      Getting a loan   
873        Taking out the loan or lease   
917        Taking out the loan or lease   
922        Taking out the loan or lease   
965        Taking out the loan or lease   

                                  Sub-issue  \
207                                    None   
410   Having problems with customer service   
426           Don't agree with fees charged   
436   Having problems with customer service   
481                Can't qualify for a loan   
564                                    None   
593                                    None   
633                                    None   
634                                    None   
637  Qualify for a better loan than offered   
643                Can't qualify for a loan   
645                                    None   
657  Qualify for a better loan than offered   
787                                    None   
811  Qualify for a better loan than offered   
826  Qualify for a better loan than offered   
834  Qualify for a better loan than offered   
873                                    None   
917                                    None   
922                                    None   
965                                    None   

                          Consumer complaint narrative  \
207  Issue with loan process and how things are han...   
410  We applied to re-finance my wife 's student lo...   
426  On XXXX/XXXX/2015 we made a payment from our X...   
436                                                NaN   
481                                                NaN   
564  Received a mail offer from SoFi of up to {$980...   
593                            

In [15]:
df.query("`Consumer disputed?` == 'No'")

Date received                  Product                 Sub-product  \
151      06/08/16             Student loan    Non-federal student loan   
168      11/23/15            Consumer Loan            Installment loan   
298      11/05/16            Consumer Loan            Installment loan   
300      11/01/16          Debt collection                 Credit card   
384      10/27/15            Consumer Loan            Installment loan   
..            ...                      ...                         ...   
921      12/05/15            Consumer Loan            Installment loan   
924      09/15/15  Bank account or service  Other bank product/service   
940      09/27/14             Student loan    Non-federal student loan   
953      02/04/15             Student loan    Non-federal student loan   
966      03/10/15            Consumer Loan            Installment loan   

                                       Issue                 Sub-issue  \
151                           Getting a loan  Can't qualify for a loan   
168             Shopping for a loan or lease                      None   
298             Taking out the loan or lease                      None   
300    Cont'd attempts collect debt not owed          Debt is not mine   
384             Taking out the loan or lease                      None   
..                                       ...                       ...   
921             Taking out the loan or lease                      None   
924  Account opening, closing, or management                      None   
940                           Getting a loan  Can't qualify for a loan   
953                           Getting a loan  Can't qualify for a loan   
966             Taking out the loan or lease                      None   

                          Consumer complaint narrative  \
151  I went to SoFi web page to see what rates woul...   
168                                                NaN   
298  I would not recommend any type of loan with th...   
300  MY MAILBOX SYSTEM HAD ALOT OF POSTAL ISSUES AN...   
384                                                NaN   
..                                                 ...   
921  This concerns a load from SoFi. I filed out al...   
924  Sofi has ignored my request to stop sending me...   
940                                                NaN   
953                                                NaN   
966                                                NaN   

    Company public response               Company State ZIP code  \
151                    None  Social Finance, Inc.    DC    95966   
168                    None  Social Finance, Inc.    NY    13211   
298                    None  Social Finance, Inc.    TX    75219   
300                    None  Social Finance, Inc.    CA    92707   
384                    None  Social Finance, Inc.    ME    04901   
..                      ...                   ...   ...      ...   
921                    None  Social Finance, Inc.    NC    28139   
924                    None  Social Finance, Inc.    TX    78552   
940                    None  Social Finance, Inc.    OR    97116   
953                    None  Social Finance, Inc.    MI    48073   
966                    None  Social Finance, Inc.    CA    94112   

               Tags Consumer consent provided? Submitted via  \
151            None           Consent provided           Web   
168            None                        NaN      Referral   
298  Older American           Consent provided           Web   
300            None           Consent provided           Web   
384            None       Consent not provided           Web   
..              ...                        ...           ...   
921            None           Consent provided           Web   
924            None           Consent provided           Web   
940   Servicemember                        NaN           Web   
953            None                        NaN           Web   
966     

Final Analysis: Out of the 992 complaints, 85 responded to the disputed section. 21 said they disputed the company's resolution of their complaints. 